manifests/resourcegroup-crd.yaml

```diff
@@ -58,7 +58,7 @@ spec:
           metadata:
             type: object
           spec:
-            description: spec defines the desired state of ResourceGroup
+            description: ResourceGroupSpec defines the desired state of ResourceGroup
```

Comment:
I'm not sure whether this is how we want descriptions to read on the CRD. I think the previous descriptions may read better. I'm hoping this will spark some discussion

/assign @karlkfi

Reply:
pretty sure we have nolint set on other api structs to avoid this.

Reply:
We generally don't - we have other API fields which surface the name of the go struct on the CRD.

But I added some nolint comments here to prevent any updates to the CRD with this change.